lib/explorer/data_frame.ex

```diff
@@ -501,6 +501,22 @@ defmodule Explorer.DataFrame do
 
   ## Conversion
 
+  @doc """
+  Converts the dataframe to the lazy version of the current backend.
+
+  If already lazy, this is a noop.
+  """
+  @spec to_lazy(df :: DataFrame.t()) :: DataFrame.t()
+  def to_lazy(df), do: Shared.apply_impl(df, :to_lazy)
+
+  @doc """
+  This collects the lazy data frame into an eager one, computing the query.
+
+  If already eager, this is a noop.
+  """
+  @spec collect(df :: DataFrame.t()) :: {:ok, DataFrame.t()} | {:error, term()}
+  def collect(df), do: Shared.apply_impl(df, :collect)
+
```

Comment:
We also want to add `def lazy()`, that simply returns the lazy version of the current backend. :)

Reply:
Oh I implemented it but then didn't expose it 😬

Reply:
Oh now I'm questioning myself. When exposing it (not defining in the behaviour), should it be `lazy/0` or `lazy/1`? If `lazy/0` is it just the lazy version of the _default_ backend? It would need to be `lazy/1`, taking a data frame, to return the lazy version of a given DF's backend.

Reply:
>  If lazy/0 is it just the lazy version of the default backend? 

Yes!

My rationale was that `lazy()` returns the lazy version of the new dataframe. So we would do:

    Explorer.DataFrame.lazy().read_csv()

And then `to_lazy` is the one converting an existing dataframe to lazy. We could have both as `lazy/0` and `lazy/1` but I thought the different names could help? 🤔

Reply:
Oh shit, I just realized that this does not quite work. I need to think about it. Let's postpone lazy() then as well. :D

Reply:
Ok... one way I think the above can work is for us to have something like:

    require Explorer.Backend.DataFrame
    lazy_delegates!()

And that will define `read_csv` and friends that call `Explorer.DataFrame` passing the backend as an option.

Is this worth it? I would say yes. I think `Explorer.DataFrame.lazy().read_csv()` is a really nice API for reading lazy dataframes. If you agree, I can try to work on that on a follow up PR. :) So you can skip `lazy/0` on this one still!

Reply:
I agree. I think that's a really nice API as well, totally. An alternative that comes to mind is to pass `lazy: true` or `:backend` as options, but I think I prefer what you suggested.

Reply:
Oh, I like the `lazy: true` option. Definitely better than backend for this use case. I feel `.lazy()` is more explicit though when reading the code (but also a bit more magic).